src/api/paginateRequestFilter.tsx

```diff
@@ -0,0 +1,16 @@
+import { IPaginateRequest } from "../interfaces/IPaginateRequest";
+
+const paginateRequestFilter = async (pageNumber: number, name: string) => {
+  try {
+    const response = await fetch(
+      `https://rickandmortyapi.com/api/character/?page=${pageNumber}&name=${name}`
+    );
+    const data: IPaginateRequest = await response.json();
+
+    return data;
+  } catch (error) {
+    console.error("Error fetching data:", error);
+  }
+};
```

Comment:
You can create class that includes several static methods. 
You could add to it your "paginateRequest" and "paginateRequestFilter"

Reply:
Even you can combine 'paginateRequest's and 'paginateRequestFilter's logic into one function. 
Because I can see main logic is the same in both functions